src/components/listing-components/authentication-utils.tsx

```diff
@@ -0,0 +1,141 @@
+import strings from "localization/strings";
+
+/**
+ * Interface for site props
+ */
+interface Site {
+  id: string;
+  name: string;
+  url: string;
+  token: string;
+}
+
+/**
+ * Interface for login params from login-dialog.tsx
+ */
+interface LoginParams {
+  selectedSite: Site | undefined;
+  username: string;
+  password: string;
+}
+
+/**
+ * Interface for refreshing token on login-dialog.tsx side
+ */
+interface TokenParams {
+  selectedSite: Site | undefined;
+  refreshToken: string;
+}
+
+/**
+ * Interface for AuthenticationResponse
+ */
+interface AuthenticationResponse {
+  accessToken: string;
+  refreshToken: string;
+  error: string;
+}
+
+/**
+ * Main component to handle 3rd party login
+ * @param params 
+ * @returns 
+ */
+const authenticate = async (params: LoginParams): Promise<AuthenticationResponse> => {
```

Comment:
```suggestion
/**
 * Main component to handle 3rd party login
 *
 * @param params 
 * @returns 
 */
const authenticate = async (params: LoginParams): Promise<AuthenticationResponse> => {
```

Reply:
Here you could add description that this authentication currently only supports kiertoon.fi-site login.